pinot-distribution/pinot-source-assembly.xml

```diff
@@ -0,0 +1,47 @@
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+<assembly
+    xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2 http://maven.apache.org/xsd/assembly-1.1.2.xsd">
+  <id>source-release</id>
+  <formats>
+    <format>tar.gz</format>
+  </formats>
```

Comment:
I actually considered that. However, that will cause the issue when people try to build the binary tarbell from the released source code since the source release won't contain `-binary` files. Spark seemed to have the same issue and their decision is to remove `-binary` files from the source release. Side effect is that people cannot build a completely valid binary release from the released source code. https://github.com/apache/spark/pull/22840

On the other hand, I checked that Apache Flink is using similar approach as Spark (dividing -binary) and they include `-binary` files to their source code. Can we downloaded from https://www.apache.org/dyn/closer.lua/flink/flink-1.7.1/flink-1.7.1-src.tgz

I would prefer the latter since it allows our users to build a valid binary tarbell (with correct LICENSE, NOTICE). 

@felixcheung How do you think?